code/model/modelreplace.h

```diff
@@ -90,6 +90,35 @@ class VirtualPOFOperationAddSpecialSubsystem : public VirtualPOFOperation {
 	void process(polymodel* pm, model_read_deferred_tasks& deferredTasks, model_parse_depth depth, const VirtualPOFDefinition& virtualPof) const override;
 };
 
+class VirtualPOFOperationAddWeapons : public VirtualPOFOperation {
+	int sourcebank, destbank = -1;
+	bool primary;
+	SCP_string appendingPOF;
+public:
+	VirtualPOFOperationAddWeapons();
+	void process(polymodel* pm, model_read_deferred_tasks& deferredTasks, model_parse_depth depth, const VirtualPOFDefinition& virtualPof) const override;
+};
+
+class VirtualPOFOperationAddDockPoint : public VirtualPOFOperation {
+	SCP_string sourcedock;
+	tl::optional<SCP_string> renameDock;
+	tl::optional<SCP_string> targetParentSubsystem;
```

Comment:
It doesn't matter, since tl::optional is a class type, we're guaranteed that it calls a viable constructor, which is the nullopt constructor for the optional